File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string new_s= "";
    for (char letter: s) {
        if ((int(letter) >= 97 && int(letter) <= 122) || (int(letter) >= 48 && int(letter) <= 57)) {
            new_s += letter;
        } else if ((int(letter) >= 65 && int(letter) <= 90)) {
            new_s += char(int(letter) + 32);
        }
    }

    string reversed_s= "";
    for (int i= new_s.length() - 1; i >= 0; i--) {
        reversed_s += new_s[i];
    }
        if (new_s != reversed_s) {
            return false;
        }
        return true;
    }
};

File: Data Structures & Algorithms/palindrome-partitioning/submission-0.py
class Solution:
    def partition(self, s: str) -> List[List[str]]:
        res = []
        parts = []
        
        def is_palindrome(s):
            return s == s[::-1]

        def dfs(start):
            if start == len(s):
                res.append(parts.copy())
                return

            for end in range(start + 1, len(s) + 1):
                part = s[start:end]

                if not is_palindrome(part):
                    continue

                parts.append(part)
                dfs(end)
                parts.pop()

        dfs(0)
        return res
            